src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -7,17 +7,18 @@
 using Microsoft.CodeAnalysis.Editing;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.SymbolCategorization;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
 {
     internal class SymbolSpecification
     {
         public Guid ID { get; private set; }
         public string Name { get; private set; }
-        public List<SymbolKindOrTypeKind> ApplicableSymbolKindList { get; private set; }
-        public List<AccessibilityKind> ApplicableAccessibilityList { get; private set; }
-        public List<ModifierKind> RequiredModifierList { get; private set; }
-        public List<string> RequiredCustomTagList { get; private set; }
+        public IEnumerable<SymbolKindOrTypeKind> ApplicableSymbolKindList { get; private set; }
+        public IEnumerable<AccessibilityKind> ApplicableAccessibilityList { get; private set; }
+        public ModifierKindEnum RequiredModifierList { get; private set; }
+        public IEnumerable<string> RequiredCustomTagList { get; private set; }
 
```

Comment:
Everything thats called a list here is actually a list of something, except `RequiredModifierList`. Consider changing that.